src/main/java/nextstep/fp/StreamStudy.java

```diff
@@ -27,7 +28,15 @@ public static void printLongestWordTop100() throws IOException {
                 .get("src/main/resources/fp/war-and-peace.txt")), StandardCharsets.UTF_8);
         List<String> words = Arrays.asList(contents.split("[\\P{L}]+"));
 
-        // TODO 이 부분에 구현한다.
+        List<String> result = words.stream()
+                .filter(w -> w.length() > 12)
+                .sorted(Comparator.comparingInt(String::length).reversed())
+                .distinct()
+                .limit(100)
+                .map(w->w.toLowerCase())
```

Comment:
method reference를 활용해 보시면 좋을 것 같아요
```suggestion
                .map(String::toLowerCase)
```